lib/tasks/mesh_ack_message.rake

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env ruby
+
+# frozen_string_literal: true
+
+desc "Acknowledge message MESH, removing it from inbox"
+task "mesh:ack_message", [:message] => :environment do |_, args|
+  message = args[:message]
+  base_url = Settings.mesh.base_url
+  mailbox = Settings.mesh.mailbox
+  authorisation = MESH.generate_authorisation
+
+  headers = {
+    "Accept" => "application/vnd.mesh.v2+json",
+    "Authorization" => "#{MESH::SCHEMA} #{authorisation}"
+  }
+
+  conn =
+    Faraday.new do |faraday|
+      faraday.ssl[:verify] = false if Rails.env.development?
+      faraday.headers = headers
+    end
+
+  response =
+    conn.put(
+      "#{base_url}/messageexchange/#{mailbox}/inbox/#{message}/status/acknowledged"
+    )
```

Comment:
I'm wondering if there's a way we can define a shared Faraday connection instance for MESH using the `base_url` which can then be used in multiple places, perhaps in the `MESH` module.


```rb
def self.connection
  @connection ||= Faraday.new(Settings.mesh.base_url, headers: {
    "Accept" => "application/vnd.mesh.v2+json",
    "Authorization" => "#{MESH::SCHEMA} #{authorisation}"
  }, ssl: { verify: Rails.env.production? })
end
```

Then in this task all you'd need is:

```rb
response = MESH.connection.put("/messageexchange/#{mailbox}/inbox/#{message}/status/acknowledged")
```

And it would reduce the duplication we've got across the `MESH` module, and these Rake tasks.

Reply:
Ya, this makes sense. 👍